apps/server/src/shared/domain/rules/submission.rule.spec.ts

```diff
@@ -254,7 +254,7 @@ describe('SubmissionRule', () => {
 						const { user, submission, permission } = setup();
 
 						const result = submissionRule.hasPermission(user, submission, {
-							action: Actions.write,
+							action: Actions.read,
```

Comment:
Some line at top 

```
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const hasWriteAccessSpy = jest.spyOn(SubmissionRule.prototype as any, 'hasWriteAccess');
hasWriteAccessSpy.mockReturnValue(true);
```

Can be writen as 

```
const hasWriteAccessSpy = jest.spyOn(SubmissionRule, 'hasWriteAccess').mockReturnValue(true);
```

Reply:
This won't work because it's a private method. I found the hack on stackoverflow and was unsure if it's okay to use it but then I saw you used it in task.entity.spec.ts already. Still I would be interested in discussing how we want to mock private methods.

Reply:
Or wait... The private modifier is just typescript! So it should work...